x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -560,26 +561,21 @@ private boolean isNoop(
             final List<RoleDescriptor> currentRoleDescriptors = parseRoleDescriptorsBytes(
                 request.getId(),
                 apiKeyDoc.roleDescriptorsBytes,
-                RoleReference.ApiKeyRoleType.ASSIGNED
+                false
             );
             if (false == (newRoleDescriptors.size() == currentRoleDescriptors.size()
-                && Set.copyOf(newRoleDescriptors).containsAll(new HashSet<>(currentRoleDescriptors)))) {
```

Comment:
The `HashSet` bit here is actually redundant: we check that the list sizes are equal and that one collection (as a set) contains the other.

Reply:
Indeed. My suggestion on the previous PR did not have it.

Reply:
https://github.com/elastic/elasticsearch/pull/88346#discussion_r918543380